components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1479,25 +1479,36 @@ void RewardsServiceImpl::SetCatalogIssuers(const std::string& json) {
 }
 
 std::pair<uint64_t, uint64_t> RewardsServiceImpl::GetEarningsRange() {
-  auto one_day_in_seconds = base::Time::kMicrosecondsPerDay /
-      base::Time::kMicrosecondsPerSecond;
-
   auto now = base::Time::Now();
   base::Time::Exploded exploded;
   now.LocalExplode(&exploded);
 
-  uint64_t offset_in_seconds;
-  if (exploded.day_of_month > 5) {
-    offset_in_seconds = (exploded.day_of_month - 5) * one_day_in_seconds;
-  } else {
-    offset_in_seconds = exploded.day_of_month * one_day_in_seconds;
+  if (exploded.day_of_month < 5) {
+    exploded.month--;
+    if (exploded.month < 1) {
+      exploded.month = 12;
+
+      exploded.year--;
+    }
   }
 
-  auto now_in_seconds = (now - base::Time()).InSeconds();
-  uint64_t from_timestamp = now_in_seconds - offset_in_seconds;
-  uint64_t to_timestamp = now_in_seconds;
+  exploded.day_of_month = 1;
+
+  exploded.hour = 0;
+  exploded.minute = 0;
+  exploded.second = 0;
```

Comment:
exploded.millisecond will probably have garbage in it if you don't explicitly zero it out, though not sure if it strictly matters here.

Reply:
good catch will fix just incase a user redeems a token before 1 second passes as this redemption would not be shown in this edge case 👍